### PR TITLE
Fetcher utilities for raw and preprocessed data from EBRAINS

### PR DESCRIPTION
**Pull request by man-shu**
This PR aims to add utilities to the API to fetch raw and preprocessed data from EBRAINS. This would require the following:

- [x] Creating a database of all the files that are available on EBRAINS, one for raw and preprocessed datasets each. This is a one-time effort and the code does not need to be made public with the API.
- [x] Adding functions/utilities to `utils.py` to use these database files and allow file-wise downloading of the raw and preprocessed datasets (when `data_type = "raw"` and `data_type = "preprocessed"`).
- [x] Update `example.py` and `get_data.ipynb` accordingly

We later moved all the database files from the docs repo to this API repo. So, all database csv files will be shipped with the fetcher initially and will subsequently be updated via a curl whenever the fetcher is imported. This would keep the database files up-to-date, without having to update the fetcher package.

- [x] Push the database files for all the datasets currently available on ebrains

### CODE REVIEW

**Comment by ferponcem** [comment]
Just added some lines to get `modality`, `path`, `task` and `session` columns tuned.
Next I am polishing the `task` column for the *json* files, and them I am doing the same for **preproc**.
Also, next commit I am adding comments (:

**Comment by ferponcem** [comment]
Just added some comments and removed some not-so-useful lines. 
I deliberately decided to put *docs* as `modality` for all json files, and *events* for all event.tsv files, but this is of course to be discussed, this might be the trickiest category.
There are just some few files to discuss, overall it's pretty much filled. 
Working on the preproc files now (:

**Comment by man-shu** [comment]
Awesome! That was fast! Thanks.

**Comment by ferponcem** [comment]
It's mostly just replicating what you've done.
What do you say we split in two scripts, one for each `raw` and `preproc`?

**Comment by man-shu** [comment]
I think columns `hash`, `last_modified`, `bytes` and `content_type` can be dropped.

`isnii` can be dropped and in `file_ext`,  `gz` can be replaced with `nii.gz`

Found these edge cases to discuss:

- `task` for `ses-00` files could be `Screening`
- `task` is currently NaN and could just be `NoTask` for
  - epi files in each session
  - some json files with `path` as `v3.0/{file_name}.json` i.e. just under v3.0 directory
- `subject` and `session` are NaN and could be `all` for:
  - all json files with acquisition parameters
  - README
  - participants.tsv

**Comment by man-shu** [comment]
> It's mostly just replicating what you've done. What do you say we split in two scripts, one for each `raw` and `preproc`?

I think it is good as it is because we will anyway have to remove the script before merging. So better to remove just one than two :D

**Comment by man-shu** [comment]
Found a more BIDS-compliant way to make the databases, using [pybids](https://github.com/bids-standard/pybids).

- `create_db.py` now makes databases for the contrasts as well. So all the columns are consistent across all IBC datasets.
- `ibc.json` is the config file used by pybids to parse filenames and extract all the BIDS entities. The final .csv files would have the columns defined under `name`, each extracted by the respective regex defined under `pattern` in this file.

**Comment by man-shu** [comment]
As discussed recently, I've moved all the ebrains-related data files here. They will be shipped with the package initially but updated via a curl command whenever it is imported. 

Like this, the user would not have to update the package repeatedly to get up-to-date info.

We are now ready for a review!

File: src/ibc_api/metadata.py [hunk 1]
```diff
@@ -62,8 +62,18 @@ def select_dataset(data_type, metadata=None):
         raise KeyError(
             f"Dataset type {data_type} not found in IBC collection."
         )
-    latest_version = _find_latest_version(dataset)
-    dataset = dataset[latest_version]
+    if version is not None:
+        version = version - 1
```
Comment (bthirion) [inline src/ibc_api/metadata.py:66]: Why do you do that ? Please provide comments

File: README.md [hunk 2]
```diff
@@ -16,21 +16,21 @@ pip install git+https://github.com/individual-brain-charting/api.git#egg=ibc_api
...
-# Returns a link that takes you to EBRAINS login page and stores an access token locally
-ibc.authenticate()
-
 # Fetch info on all available files
 # Load as a pandas dataframe and save as ibc_data/available_{data_type}.csv 
 db = ibc.get_info(data_type="volume_maps")
 
 # Keep statistic maps for sub-08, for task-Discount
-filtered_db = ibc.filter_data(db, subject_list=["sub-08"], task_list=["Discount"])
+filtered_db = ibc.filter_data(db, subject_list=[8], task_list=["Discount"])
```
Comment (bthirion) [inline README.md]: Why have you changes subjects from strings to int ? I find it less logical/readable/BIDS friendly.

**Review by bthirion — COMMENTED** [review]
LGTM otherwise.

File: README.md (hunk 2 above)
Comment (man-shu) [reply README.md]: Actually, pybids creates `subject` column in the dataframe with subject numbers (and session numbers, run numbers) as int. So I had to make that change to accommodate for that.

Comment (bthirion) [reply README.md]: OK. I think that this is a bid idea. Maybe leave a comment in the code to notify that.

Comment (man-shu) [reply README.md]: Ok, so changing int to str of double-digit number with leading zeros

File: src/ibc_api/utils.py [hunk 3]
```diff
@@ -115,7 +128,7 @@ def filter_data(db, subject_list=SUBJECTS, task_list=False):
         dataframe with information about all files in the dataset
     subject_list : list, optional
         list of subjects to keep, by default SUBJECTS, SUBJECTS contains all
-        subjects from sub-01 to sub-15
+        subjects from 1 to 15
```
Comment (bthirion) [inline src/ibc_api/utils.py]:
```suggestion
        subjects from ["01", "02", "04",...,"15"]
```

File: src/ibc_api/utils.py [hunk 4]
```diff
@@ -145,59 +158,34 @@ def get_file_paths(db, metadata=METADATA):
 
     Returns
     -------
-    list
-        list of file paths for each file in the input dataframe
+    list, list
```
Comment (bthirion) [inline src/ibc_api/utils.py]:
```suggestion
    filenames, list
```

**Review by bthirion — COMMENTED** [review]
LGTM overall. Thx !

**Comment by man-shu** [comment]
Merging now!

**Pull request merged**